utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -69,6 +69,66 @@ fun File.unpack(targetDirectory: File) =
         inputStream().unpack(name, targetDirectory)
     }
 
+/**
+ * Unpack the [File] assuming it is a 7-Zip archive. This implementation ignores empty directories and symbolic links.
+ */
+fun File.unpack7Zip(targetDirectory: File) {
+    SevenZFile(this).use {
+        while (true) {
+            val entry = it.nextEntry ?: break
+
+            if (entry.isDirectory || entry.isAntiItem) {
+                continue
+            }
+
+            val target = targetDirectory.resolve(entry.name)
+
+            // There is no guarantee that directory entries appear before file entries, so always ensure the parent
+            // directory for a file exists.
+            target.parentFile.safeMkdirs()
+
```

Comment:
Please use the more memory friendly `InputStream.copyTo()` ?
like e.g. `entry.getInputStream().copyTo(output, BUFFER_SIZE)`